src/test/java/org/apache/mesos/specification/CustomTaskSetTest.java

```diff
@@ -0,0 +1,47 @@
+package org.apache.mesos.specification;
+
+import org.apache.mesos.Protos;
+import org.junit.Test;
+import org.testng.Assert;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class tests the ability to customize TaskTypeSpecifications.
+ */
+public class CustomTaskSetTest {
+
+    @Test
+    public void testCustomCommand() {
+        TaskSet taskSet = CustomTaskSet.create("custom", 3);
+        Assert.assertEquals("custom 0", taskSet.getTaskSpecifications().get(0).getCommand().getValue());
+        Assert.assertEquals("custom 2", taskSet.getTaskSpecifications().get(2).getCommand().getValue());
+
+        Assert.assertEquals("custom_0", taskSet.getTaskSpecifications().get(0).getName());
+        Assert.assertEquals("custom_2", taskSet.getTaskSpecifications().get(2).getName());
+    }
+
+    public static class CustomTaskSet extends DefaultTaskSet {
```

Comment:
@nickbp @mrbrowning:  Here's an example of providing your own custom TaskSpecifications.  What do you think?  This change also fixes the weirdness around `DefaultPhaseSpecification` always ending up being composed of `DefaultTaskSpecification`s